app/src/test/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpassTest.kt

```diff
@@ -1279,6 +1330,23 @@ class HikeRoutesRepositoryOverpassTest {
     assert(onSuccessCalled)
   }
 
+  @Test
+  fun getRoutes_worksOnCrossingTheDateLine() {
+    val mockCall = mock(Call::class.java)
+    `when`(mockClient.newCall(any())).thenReturn(mockCall)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    `when`(mockCall.enqueue(callbackCapture.capture())).then {
+      callbackCapture.firstValue.onResponse(mockCall, responseCrossingTheDateLine)
+    }
+
+    hikingRouteProviderRepositoryOverpass.getRoutes(
+        bounds, { routes -> assertEquals(0, routes.size) }) {
+          fail("Failed to fetch routes from Overpass API")
+        }
```

Comment:
Suggestion: make sure that one of the callbacks is called.

```suggestion
    var successCalled = false
    hikingRouteProviderRepositoryOverpass.getRoutes(
        bounds, { routes -> assertEquals(0, routes.size); successCalled = true }) {
          fail("Failed to fetch routes from Overpass API")
        }
        
    assertTrue(successCalled)
```

Reply:
Nice catch!